### PR TITLE
OpenLayers 5

### PR DESCRIPTION
**Pull request by samuel-girard**
Hi,
I'm working on the integration with OpenLayers 5.
All the work will be done on this branch.

Hope to have something to show within the next few days.

### CODE REVIEW

**Comment by davinkevin** [comment]
Can you rebase it on the branch [next](https://github.com/quentin-ol/ngx-openlayers/tree/next) ?

**Comment by samuel-girard** [comment]
Hello
I just rebased it on [next](https://github.com/quentin-ol/ngx-openlayers/tree/next) branch

**Comment by davinkevin** [comment]
I still have the change done by @Neonox31 in the pull request which is not coherent with a rebase upon `next` branches. Could you check it?

BTW, I've seen a lot of style modification in your code (mainly around the @Input on two line instead of one). Could you follow what has already be done to simplify the merge request. As is, it seems to be too many modification to still be clear.

Thanks

**Comment by samuel-girard** [comment]
OK, I previously only rebased the branch, but not the PR.  
It is OK now, you shouldn't see any more of these style modifications.  

For information, this is not yet ready to be merged: it is working well as it is (no issue found so far), but it lacks of OpenLayers typings. I temporarily created an ol-models to workaround this problem.  
There is a big work beeing done in OpenLayers for that, so I hopefully can fix this soon.

**Comment by FallenRiteMonk** [comment]
what is the status of this?

**Comment by samuel-girard** [comment]
Hello,
Updated after the release of OpenLayers 5.3, this branch can now be merged.  
There are still typings issues: for example your IDE will not offer you autocompletion for OpenLayers types used inside your application, but at least, ngx-openlayers will not fail to compile.  

There is a workaround for the autocompletion if it is a big issue: https://github.com/openlayers/openlayers/issues/8448#issuecomment-441777821

**Comment by samuel-girard** [comment]
Hi @Neonox31 
I rebased this branch on `next`. You can merge it.
Thanks

**Comment by HarelM** [comment]
Any updates on this? It would really help to have ol 5 in next branch and pre-released to npm.

**Comment by Yakoust** [comment]
@kekel87 @davinkevin @Neonox31 could you please have a look on this PR?

**Comment by kekel87** [comment]
There are still `openlayers` imports in `attribution.component.ts` and `attributions.component.ts`.

File: projects/ngx-openlayers/src/lib/interactions/draw.component.ts [hunk 1]
```diff
@@ -1,65 +1,71 @@
...
   @Input()
-  type: ol.geom.GeometryType;
+  type: GeometryType;
   @Input()
   maxPoints?: number;
   @Input()
   minPoints?: number;
   @Input()
-  finishCondition?: ol.EventsConditionType;
+  finishCondition?: any; // TODO: ol.EventsConditionType;
```
Comment (Yakoust) [inline projects/ngx-openlayers/src/lib/interactions/draw.component.ts]: In ol 5.3.0 finishCondition has type Condition from import { Condition } from 'ol/events/condition';

see https://openlayers.org/en/latest/apidoc/module-ol_interaction_Draw.html

Comment (Yakoust) [reply projects/ngx-openlayers/src/lib/interactions/draw.component.ts]: Same below

File: projects/ngx-openlayers/src/lib/sources/imagearcgisrest.component.ts [hunk 2]
```diff
@@ -12,25 +14,25 @@ import { SourceComponent } from './source.component';
...
 
   @Input() projection: ProjectionLike | string;
   @Input() url: string;
-  @Input() attributions: Attribution[];
+  @Input() attributions: AttributionLike[];
   @Input() crossOrigin?: string;
-  @Input() imageLoadFunction?: ImageLoadFunctionType;
+  @Input() imageLoadFunction?: LoadFunction;
   @Input() params?: { [k: string]: any };
   @Input() ratio = 1;
```
Comment (Yakoust) [inline projects/ngx-openlayers/src/lib/sources/imagearcgisrest.component.ts]: By default ratio is set to 1.5

https://openlayers.org/en/latest/apidoc/module-ol_source_ImageArcGISRest-ImageArcGISRest.html

File: projects/ngx-openlayers/src/lib/sources/imagearcgisrest.component.ts [hunk 3]
```diff
@@ -12,25 +14,25 @@ import { SourceComponent } from './source.component';
...
   @Input() resolutions?: number[];
-  @Input() logo?: string | olx.LogoOptions;
+  @Input() logo?: string; // (string | olx.LogoOptions);
```
Comment (Yakoust) [inline projects/ngx-openlayers/src/lib/sources/imagearcgisrest.component.ts]: It seems that `logo` is no more a property of ImageArcGISRest

https://openlayers.org/en/latest/apidoc/module-ol_source_ImageArcGISRest.html

File: projects/ngx-openlayers/src/lib/sources/imagewms.component.ts [hunk 4]
```diff
@@ -32,9 +36,9 @@ export class SourceImageWMSComponent extends SourceComponent implements OnChange
   @Input()
   serverType: string;
   @Input()
-  imageLoadFunction?: ImageLoadFunctionType;
+  imageLoadFunction?: LoadFunction;
   @Input()
-  logo: string | olx.LogoOptions;
+  logo: string; // TODO: (string | olx.LogoOptions);
```
Comment (Yakoust) [inline projects/ngx-openlayers/src/lib/sources/imagewms.component.ts]: logo property does not exist anymore on ol 5.3.0

https://openlayers.org/en/latest/apidoc/module-ol_source_ImageWMS.html

File: projects/ngx-openlayers/src/lib/sources/tilewmts.component.ts [hunk 5]
```diff
@@ -58,11 +62,11 @@ export class SourceTileWMTSComponent extends SourceComponent implements AfterCon
   @Input()
   matrixSet: string;
   @Input()
-  dimensions?: GlobalObject;
+  dimensions?: Object; // TODO: GlobalObject;
```
Comment (Yakoust) [inline projects/ngx-openlayers/src/lib/sources/tilewmts.component.ts]: that // TODO: should be removed

File: projects/ngx-openlayers/src/lib/sources/tilewmts.component.ts [hunk 6]
```diff
@@ -21,34 +25,34 @@ import { TileGridWMTSComponent } from '../tilegridwmts.component';
...
 export class SourceTileWMTSComponent extends SourceComponent implements AfterContentInit, OnChanges {
-  instance: source.WMTS;
+  instance: WMTS;
   @Input()
   cacheSize?: number;
   @Input()
   crossOrigin?: string;
   @Input()
-  logo?: string | olx.LogoOptions;
+  logo?: string; // TODO: string | olx.LogoOptions;
```
Comment (Yakoust) [inline projects/ngx-openlayers/src/lib/sources/tilewmts.component.ts]: logo property does not exist anymore on ol 5.3.0

File: projects/ngx-openlayers/src/lib/sources/xyz.component.ts [hunk 7]
```diff
@@ -41,15 +43,15 @@ export class SourceXYZComponent extends SourceComponent implements AfterContentI
   @Input()
   maxZoom: number;
   @Input()
-  tileGrid: tilegrid.TileGrid;
+  tileGrid: any;
```
Comment (Yakoust) [inline projects/ngx-openlayers/src/lib/sources/xyz.component.ts]: I think that this shoul be `tileGrid: TileGrid;` with   `import { TileGrid } from 'ol/tilegrid';`

File: projects/ngx-openlayers/src/lib/sources/xyz.component.ts [hunk 8]
```diff
@@ -41,15 +43,15 @@ export class SourceXYZComponent extends SourceComponent implements AfterContentI
...
   @Input()
-  tileLoadFunction?: TileLoadFunctionType;
+  tileLoadFunction?: any;
```
Comment (Yakoust) [inline projects/ngx-openlayers/src/lib/sources/xyz.component.ts]: I think that this shoul be `tileLoadFunction: TileLoadFunctionType;` with `import { LoadFunction } from 'ol/Tile';`

File: projects/ngx-openlayers/src/lib/sources/xyz.component.ts [hunk 9]
```diff
@@ -41,15 +43,15 @@ export class SourceXYZComponent extends SourceComponent implements AfterContentI
...
   @Input()
   tilePixelRatio: number;
   @Input()
   tileSize: number | Size;
   @Input()
-  tileUrlFunction: TileUrlFunctionType;
+  tileUrlFunction: any;
```
Comment (Yakoust) [inline projects/ngx-openlayers/src/lib/sources/xyz.component.ts]: I think that this shoul be `tileUrlFunction: UrlFunction; with `import { UrlFunction } from 'ol/Tile';`

**Comment by Yakoust** [comment]
@samuel-girard are you available to debate/resolve discussions?

**Comment by samuel-girard** [comment]
Hello @Yakoust 
Thanks for your feedback on this PR.  
I quickly had a look at your points and it seems good to me, although I haven't checked everything in details.  
Unfortunately, I will not have time to apply the changes so I will be thankful if someone here could take care of it.

**Comment by Yakoust** [comment]
> There are still `openlayers` imports in `attribution.component.ts` and `attributions.component.ts`.

I also export those component in module. There were missing

**Comment by Yakoust** [comment]
Thanks @remiHau for your comment on https://github.com/quentin-ol/ngx-openlayers/issues/225
OL 5 version has changed to 5.3.1 in this PR

**Pull request merged**